packages/react/src/components/DatePicker/next/DatePicker.js

```diff
@@ -0,0 +1,636 @@
+          calendarContainer,
+          selectedDateElem: fpSelectedDateElem,
+          todayDateElem: fptodayDateElem,
+        } = calendar;
+        const selectedDateElem =
+          calendarContainer.querySelector('.selected') && fpSelectedDateElem;
+        const todayDateElem =
+          calendarContainer.querySelector('.today') && fptodayDateElem;
+        (
+          selectedDateElem ||
+          todayDateElem ||
+          calendarContainer.querySelector('.flatpickr-day[tabindex]') ||
+          calendarContainer
+        ).focus();
+      }
+    }
+
+    function handleOnChange() {
+      if (start.value !== '') {
+        return;
+      }
+
+      if (!calendar.selectedDates) {
+        return;
+      }
+
+      if (calendar.selectedDates.length === 0) {
+        return;
+      }
+
+      calendar.clear();
+      calendar.input.focus();
+    }
+
+    if (start) {
+      start.addEventListener('keydown', handleArrowDown);
+      start.addEventListener('change', handleOnChange);
+
+      // Flatpickr's calendar dialog is not rendered in a landmark causing an
+      // error with IBM Equal Access Accessibility Checker so we add an aria
+      // role to the container div.
+      calendar.calendarContainer.setAttribute('role', 'region');
+      // IBM EAAC requires an aria-label on a role='region'
+      calendar.calendarContainer.setAttribute(
+        'aria-label',
+        'calendar-container'
+      );
+    }
+
+    if (end) {
+      end.addEventListener('keydown', handleArrowDown);
+      end.addEventListener('change', handleOnChange);
+    }
+
+    //component did unmount equivalent
+    return () => {
+      // Note: if the `startInputField` ref is undefined then calendar will be
+      // of type: Array and `destroy` will not be defined
+      if (calendar && calendar.destroy) {
+        calendar.destroy();
+      }
+
+      if (start) {
+        start.removeEventListener('keydown', handleArrowDown);
+        start.removeEventListener('change', handleOnChange);
+      }
+
+      if (end) {
+        end.removeEventListener('keydown', handleArrowDown);
+        end.removeEventListener('change', handleOnChange);
+      }
+    };
+  }, [savedOnChange, savedOnClose, savedOnOpen]); //eslint-disable-line react-hooks/exhaustive-deps
+
+  // component did update equivalent
+  useEffect(() => {
+    changes.forEach((change) => {
+      // Coordinate when the given `value` prop changes. When this happens, we
+      // should update the calendar to the new value.
+      if (change.name === 'value') {
+        if (calendarRef.current) {
+          calendarRef.current.setDate(change.value);
+          updateClassNames(calendarRef.current);
+        } else {
+          startInputField.current.value = change.value;
+        }
+        return;
+      }
+
+      if (calendarRef.current) {
+        calendarRef.current.set(change.name, change.value);
+      }
+    });
+  }, [changes]); //eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
This block seems to use `useTrackedChanges` but that hook is currently only returning an array. Does this need to be updated for the cDU scenario from the class component?

Reply:
@joshblack I updated the PR. Tbh syncing prev & current "states" is always a little confusing for me. I was trying to wrap my head around how to best do this, and just went with something similar to how the class component did it. Lmk if this works. 😅